packages/blade/src/components/ActionList/ActionListFooter.tsx

```diff
@@ -17,6 +17,7 @@ import {
   isValidAllowedChildren,
 } from '~utils';
 import { Text } from '~components/Typography';
+import { useDropdownBottomSheetContext } from '~components/BottomSheet/BottomSheetContext';
```

Comment:
Okay so i've renamed the context. 

So two things: 
- hasBottomSheet - This is for Dropdown <> BottomSheet glue 
- isInBottomSheet - This is for BottomSheet <> ActionList glue

**hasBottomSheet:** 

I've renamed to "dropdownHasBottomSheet" and changed the context name to be more descriptive 
This flag let's dropdown knows if it's rendering a BottomSheet. By default this is false but when we render [BottomSheet we set this flag to true.](https://github.com/razorpay/blade/blob/85e14297cc0d4574dbd8a9319cdad206cb93ef0e/packages/blade/src/components/BottomSheet/BottomSheet.tsx#L180-L181)

We use this flag to prevent few behaviours which are SelectDropdown specific, for example we check this flag to [ensure that we don't close the bottomsheet](https://github.com/razorpay/blade/blob/85e14297cc0d4574dbd8a9319cdad206cb93ef0e/packages/blade/src/components/Dropdown/useDropdown.ts#L239-L241) when SelectInput get's blurred because while when we open the bottomsheet we move the focus to the close button (via initialFocusRef) 


**isInBottomSheet:** 

This is for ActionList to know if it's "inside" BottomSheet, we use this flag to do visual changes based on the flag, for example [we remove the ActionList's boxShadow if this flag is true](https://github.com/razorpay/blade/blob/85e14297cc0d4574dbd8a9319cdad206cb93ef0e/packages/blade/src/components/ActionList/styles/getBaseActionListStyles.ts#L32)